pkg/git/repository.go

```diff
@@ -312,14 +319,44 @@ func (r *repo) commit(opts *PushOptions) (*plumbing.Hash, error) {
 		}
 	}
 
-	h, err = w.Commit(opts.CommitMsg, &gg.CommitOptions{All: true})
+	h, err = w.Commit(opts.CommitMsg, &gg.CommitOptions{
+		All:    true,
+		Author: author,
+	})
 	if err != nil {
 		return nil, err
 	}
 
 	return &h, nil
 }
 
+func (r *repo) getAuthor(ctx context.Context, providerType string) (*object.Signature, error) {
+	username, email, err := r.provider.GetAuthor(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get commiter data. Error: %w", err)
```

Comment:
`commiter` -> `committer`

Reply:
Usually you should not do: `fmt.Errorf("something. Error: %w")` you just do `fmt.Errorf("something: %w", err)`.

Reply:
Should we fail to commit if we don't get the author from the provider? Does it make sense the a provider will not return an error but will have an empty username/email?

Reply:
currently, github/gitlab/gitea should all return valid username/email from the token (or fail with an error).
but in ado, it just returns `"", "", nil`, and then the user will get the message about missing information in git config file, because this will be the only way they can get it to work.